packages/entity/src/EntityMutator.ts

```diff
@@ -331,11 +335,109 @@ export class DeleteMutator<
 
     return result();
   }
+
+  /**
+   * Finds all entities referencing specified entity and either deletes, nullifies, or
+   * invalidates the cache depending on the {@link OnDeleteBehavior} of the field referencing
+   * the specified entity.
+   *
+   * @remarks
+   * This works by doing reverse fan-out queries:
+   * 1. Load all entity configurations of entity types that reference this type of entity
+   * 2. For each entity configuration, find all fields that contain edges to this type of entity
+   * 3. For each edge field, load all entities with an edge from target entity to this entity via that field
+   * 4. Perform desired OnDeleteBehavior for entities
+   *
+   * @param entity - entity to find all references to
+   */
+  private async processEntityDeletionForInboundEdgesAsync(entity: TEntity): Promise<void> {
+    const inboundEdges = this.entityConfiguration.inboundEdges;
+    await Promise.all(
+      inboundEdges.map(async (entityClass) => {
+        return await mapMapAsync(
+          entityClass.getCompanionDefinition().entityConfiguration.schema,
+          async (fieldDefinition, fieldName) => {
+            const association = fieldDefinition.association;
+            if (!association) {
+              return;
+            }
+
+            const associatedConfiguration = association.associatedEntityClass.getCompanionDefinition()
+              .entityConfiguration;
+            if (associatedConfiguration !== this.entityConfiguration) {
+              return;
+            }
+
+            const associatedEntityLookupByField = association.associatedEntityLookupByField;
+            let inboundReferenceEntities: readonly ReadonlyEntity<any, any, any, any>[];
+
+            const loaderFactory = this.viewerContext
+              .getViewerScopedEntityCompanionForClass(entityClass)
+              .getLoaderFactory();
+            const mutatorFactory = this.viewerContext
+              .getViewerScopedEntityCompanionForClass(entityClass)
+              .getMutatorFactory();
+
+            if (associatedEntityLookupByField) {
+              inboundReferenceEntities = await loaderFactory
+                .forLoad(this.queryContext)
+                .enforcing()
+                .loadManyByFieldEqualingAsync(
+                  fieldName,
+                  entity.getField(associatedEntityLookupByField as any)
+                );
+            } else {
+              inboundReferenceEntities = await loaderFactory
+                .forLoad(this.queryContext)
+                .enforcing()
+                .loadManyByFieldEqualingAsync(fieldName, entity.getID());
+            }
+
+            switch (association.edgeDeletionBehavior) {
+              case undefined:
+              case EntityEdgeDeletionBehavior.INVALIDATE_CACHE: {
+                await Promise.all(
+                  inboundReferenceEntities.map((inboundReferenceEntity) =>
+                    loaderFactory
+                      .forLoad(this.queryContext)
```

Comment:
If we are passing in `queryContext` into the delete calls, does this mean edge deletion is always done in a transaction?